projects/ad7746-ebz/src/app/headless.c

```diff
@@ -0,0 +1,170 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/platform.h>
+#include "adi_initialize.h"
+#include <drivers/pwr/adi_pwr.h>
+
+#include "ad7746.h"
+#include "i2c.h"
+#include "delay.h"
+#include "print_log.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "uart_stdio.h"
+
+
+int32_t platform_init()
+{
+	if (ADI_PWR_SUCCESS != adi_pwr_Init())
+		return FAILURE;
+
+	if (ADI_PWR_SUCCESS != adi_pwr_SetClockDivider(ADI_CLOCK_HCLK, 1u))
+		return FAILURE;
+
+	if (ADI_PWR_SUCCESS != adi_pwr_SetClockDivider(ADI_CLOCK_PCLK, 1u))
+		return FAILURE;
+
+	adi_initComponents();
+	return 0;
+}
+
+int main(void)
+{
+	int32_t ret;
+	char input[10];
+	uint32_t capData = 0;
+	uint32_t min = 0;
+	uint32_t max = 0;
+	uint32_t mm = 0;
+	uint32_t temperature = 0;
+	struct ad7746_init_param adcip;
+	struct ad7746_dev *adc = NULL;
+
+	struct uart_desc *uart;
+	struct aducm_uart_init_param xuip = {
+		.parity = UART_NO_PARITY,
+		.stop_bits = UART_ONE_STOPBIT,
+		.word_length = UART_WORDLEN_8BITS
+	};
+	struct uart_init_param uip = {
+		.device_id = 0,
+		.baud_rate = 115200,
+		.size = UART_CS_8,
+		.parity = UART_PAR_NO,
+		.stop = UART_STOP_1,
+		.extra = &xuip,
+	};
+	struct i2c_platform_ops ops = {
+		.i2c_ops_init = i2c_init,
+		.i2c_ops_remove = i2c_remove,
+		.i2c_ops_write = i2c_write,
+		.i2c_ops_read = i2c_read,
+	};
+
+	ret = platform_init();
+	if (ret < 0)
+		goto error;
+	ret = uart_init(&uart, &uip);
+	if (ret < 0)
+		goto error;
+
+	init_uart_stdio(uart);
+
+	adcip.i2c_init = (struct i2c_init_param) {
+		.max_speed_hz = 100000,
+		.slave_address = AD7746_ADDRESS,
+		.platform_ops = &ops,
+	};
+	adcip.setup = (struct ad7746_setup) {
+		.cap = {
+			.capen = true,
+			.cin2 = true,
+			.capdiff = false,
+			.capchop = true
+		},
+		.vt = {
+			.vten = true,
+			.vtmd = AD7746_VTMD_INT_TEMP,
+			.extref = false,
+			.vtshort = false,
+			.vtchop = true
+		},
+		.exc = {
+			.clkctrl = false,
+			.excon = false,
+			.excb = AD7746_EXC_PIN_DISABLED,
+			.exca = AD7746_EXC_PIN_NORMAL,
+			.exclvl = AD7746_EXCLVL_4_DIV_8
+		},
+		.config = {
+			.vtf = 0,
+			.capf = 0,
+			.md = AD7746_MODE_CONT
+		}
+	};
+
+	pr_info("Hello!\n");
+
+	ret = ad7746_init(&adc, &adcip);
+	if (ret < 0) {
+		pr_err("AD7746 initialization failed with: %d\n", ret);
+		goto error;
+	}
+	pr_info("[INIT] AD7746 initialization ok.\n");
+
+	ret = ad7746_set_cap_dac_a(adc, true, 0x42);
+	if (ret < 0) {
+		pr_err("DACA setting failed with: %d\n", ret);
+		goto error;
+	}
+
+	pr_info("[CALIB] 1. Remove the ruler and press ENTER. ");
+	fflush(stdout);
+	scanf("%s", input);
+	ret = ad7746_get_cap_data(adc, &min);
+	if (ret < 0) {
+		pr_err("CAP reading failed with: %d\n", ret);
+		goto error;
+	}
+
+	pr_info("[CALIB] 2. Place ruler to 51mm (2inch) and press ENTER.");
+	fflush(stdout);
+	scanf("%s", input);
```

Comment:
Where input is usesd?

Reply:
this is used only as a getline() but because of the stdin issue on aducm I can't test how getline() works, whereas this scanf I know it will work when that stdin issue gets fixed